### ISSUE TITLE
I made a version of this compatible with mac OS

### ISSUE THREAD

**Issue by terwanerik** [lucavb/homebridge-magic-blue-bulb#5]
Hi!

First of all, nice job! I found your repository and noticed it didn't work because of the lack of MAC addresses for BT devices on Mac. So i rebuild your plugin to use the native `services` and `characteristics` of the BTLE lamps.

I initially figured i could keep them working side by side (Raspberry Pi / Mac), but had to change quite a lot of code to get it to work.

Here it is; https://github.com/terwanerik/homebridge-magic-blue-bulb

Maybe someone can figure out if this works on Raspberry Pi as well? Than i'd be more than happy to do a pull request :)

Cheers!

**Comment by terwanerik** [comment]
PS: an issue is not really the right tool for the job in this case, but figured it'd be good enough.

**Comment by lucavb** [comment]
First of all nice work. For me personally the MAC address was never really an issue but maybe this change opens this project up for other people. Did you by the way make any experience regarding the handle numbers?

I don’t exactly use this project any more so I can’t really tell whether it works on the Pi. From a quick glance I didn’t see any reasons why it shouldn’t. Homebridge will probably require sudo because of bluetooth but it did so before as well.

**Comment by terwanerik** [comment]
Yeah the handle numbers basically became irrelevant because i'm sending the data directly to the (writeable) `characteristic`, so that's nice. 

Alright, yeah i'd figure that too, i'm only using documented `noble` calls that should work cross-platform.

**Comment by lucavb** [comment]
So I know that quite a lot of time has passed, since we discussed this. I pretty much just released an alpha version of this plugin. I rewrote the project with TypeScript and switched out the noble dependency against the currently still maintained one. The current version is pretty much alpha since I don't really have those bulbs anymore, so I wanted to ask people to give the version 2.0.0 a try and maybe provide some feedback. I think this should also solve your problem, even though I am aware that you have forked this project and released your own version. I don't know how much you are still relying on this project, but maybe we can merge this back together with this?
